staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/aes/aes.go

```diff
@@ -23,20 +23,38 @@ import (
 	"crypto/aes"
 	"crypto/cipher"
 	"crypto/rand"
+	"crypto/sha256"
 	"encoding/binary"
 	"errors"
 	"fmt"
 	"io"
 	"sync/atomic"
 	"time"
 
+	"golang.org/x/crypto/hkdf"
+
 	"k8s.io/apiserver/pkg/storage/value"
 	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
 )
 
+// commonSize is the length of various security sensitive byte slices such as encryption keys.
+// Do not change this value.  It would be a backward incompatible change.
+const commonSize = 32
```

Comment:
nit: Break this constant up into keySize, seedSize, kdfNonceSize, etc... even if they are all 32 so that they are easier to make sense of in context.

Right now reading code like:
```
info := data[:commonSize]
```
Has a decent amount of cognitive overhead.